src/chain_sync/tipset_syncer.rs

```diff
@@ -1072,61 +1085,40 @@ async fn sync_messages_check_state<DB: Blockstore + Clone + Send + Sync + 'stati
     genesis: &Tipset,
     invalid_block_strategy: InvalidBlockStrategy,
 ) -> Result<(), TipsetRangeSyncerError<C>> {
-    let task_chainstore = chainstore.clone();
     let request_window = state_manager.chain_config().request_window;
-    // Spawn a background task for the chain_exchange message requests
-
-    let (s, r) = flume::bounded(request_window * 4);
-    let handle = tokio::task::spawn(async move {
-        let mut batch: Vec<Arc<Tipset>> = Vec::with_capacity(request_window);
-
-        // Visit tipsets in chronological order
-        for tipset in tipsets.into_iter().rev() {
-            // If the current tipset batch is empty and we already have the
-            // messages for the current tipset, skip the download and send
-            // it directly to the validator.
-            if batch.is_empty() {
-                if let Some(full_tipset) = task_chainstore.fill_tipset(&tipset) {
-                    s.send_async(full_tipset).await?;
-                    continue;
-                }
+    let db = chainstore.blockstore();
+
+    // Stream through the tipsets from lowest epoch to highest epoch
+    stream::iter(tipsets.into_iter().rev())
+        // Chunk tipsets in batches (default batch size is 8)
```

Comment:
Tell me more about what this means?

We ask peers about N tipsets. This is a batch...
If N is too big, the peer will hang up...

Reply:
Elaborated on batch sizes in `fetch_batch`.